includes/class-database-cleanup.php

```
@@ -189,19 +259,26 @@ public static function clean_trashed_posts() {
-		$result = $wpdb->query( "DELETE FROM $wpdb->comments WHERE comment_approved = 'spam'" );
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Direct SQL is necessary for efficient bulk cleanup and caching is not required for one-off delete operations.
+			$comments_deleted = $wpdb->query( "DELETE FROM $wpdb->comments WHERE comment_approved = 'spam' LIMIT 1000" );
 
-		return $result;
+			if ( $comments_deleted ) {
+				$deleted += $comments_deleted;
+			}
+		} while ( $comments_deleted > 0 || $meta_deleted > 0 );
+
+		return $deleted;
 	}
```

PHPMD (2.15.0)

[error] The method clean_spam_comments is not named in camelCase. (undefined)

(CamelCaseMethodName)